spirv_msl.hpp

```diff
@@ -115,22 +116,26 @@ class CompilerMSL : public CompilerGLSL
 	                             uint32_t coord, uint32_t coord_components, uint32_t dref, uint32_t grad_x,
 	                             uint32_t grad_y, uint32_t lod, uint32_t coffset, uint32_t offset, uint32_t bias,
 	                             uint32_t comp, uint32_t sample, bool *p_forward) override;
-	std::string clean_func_name(std::string func_name) override;
 
 	void preprocess_op_codes();
 	void emit_custom_functions();
 	void localize_global_variables();
 	void extract_global_variables_from_functions();
+
+	std::unordered_map<uint32_t, std::unordered_set<uint32_t>> function_global_vars;
```

Comment:
I use unordered_map, because i need a pure lookup. I think output should always be the same, since i always check the key (no need to traverse the map in a certain sequence).